core/libdeno/libdeno.d.ts

```diff
@@ -37,4 +37,7 @@ declare interface DenoCore {
   evalContext(code: string): [any, EvalErrorInfo | null];
 
   errorToJSON: (e: Error) => string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setGCObserver: <T>(o: T, callback: (o: T) => void) => T;
```

Comment:
I think this would be better named "setWeakCallback"

I also think this would be a good place for some JSDOC. The docs from v8.h should be forwarded here:
```js
/** Install a finalization callback on this object.
   *  NOTE: There is no guarantee as to *when* or even *if* the callback is
   *  invoked. The invocation is performed solely on a best effort basis.
   *  As always, GC-based finalization should *not* be relied upon for any
   *  critical form of resource management!
  */
```

Reply:
Is it really so bad?
Is it guaranteed to execute prior to the program exiting?
What then _should_ / can this be used for?

Reply:
@hayd There is no guarantee about when this will be invoked. It is possible that we can force a GC on program exit, for sanity (just like how fds are closed on program exit).

That being said, this is way better than simply leaving program hang forever due to a resource not closed while losing all reference to it. Also I do remember v8 now has some [very interesting GC policies](https://v8.dev/blog/trash-talk).

Node also uses weak callbacks for cleaning up resources e.g. for the `vm` module

Reply:
Does explicit gc ensure it calls? Ry's doc/comment suggests there's an "if" here too.

I like the idea of an explicit gc call prior to exit.